docs/source/workflows/data_sources.rst

```diff
@@ -0,0 +1,90 @@
+Data Sources
+=============
+
+Data sources are used by modules to pull data from outside of the AI engine.
+For example, a :doc:`predictor <predictors>` may need to define external training data.
+
+CSV Data Source
+----------------
+
+The :class:`~citrine.informatics.data_sources.CSVDataSource` draws data from a CSV file stored on the data platform and annotates it by mapping each column name to a :class:`~citrine.informatics.descriptors.Descriptor`.
+The file is referenced via a :class:`~citrine.resources.file_link.FileLink`.
+Each FileLink references an explicit version of the CSV file.
+Uploading a new file with the same name will produce a new version of the file with a new FileLink.
+
+The columns in the CSV are extracted and parsed by a mapping of column header names to descriptors.
+Columns in the CSV that are not mapped with a descriptor are ignored.
+
+Assume that a file data.csv exists with the following contents:
+
+.. code::
+
+    Chemical Formula,Gap,Crystallinity
+    Bi2Te3,0.153,Single crystalline
+    Mg2Ge,0.567,Single crystalline
+    GeTe,0.7,Amorphous
+    Sb2Se3,1.15,Polycrystalline
+
+That file could be used as the training data for a predictor as:
+
+.. code:: python
+
+    from citrine.informatics.data_sources import CSVDataSource
+    from citrine.informatics.predictors import SimpleMLPredictor
+    from citrine.informatics.descriptors import RealDescriptor, CategoricalDescriptor, ChemicalFormulaDescriptor
+
+    file_link = dataset.files.upload("./data.csv", "bandgap_data.csv")
+
+    data_source = CsvDataSource(
+        file_link = file_link,
+        column_definitions = {
+            "Chemical Formula": ChemicalFormulaDescriptor("formula"),
+            "Gap": RealDescriptor("Band gap", lower_bound=0, upper_bound=20, units="eV"),
+            "Crystallinity": CategoricalDescriptor("Crystallinity", categories=[
+                "Single crystalline", "Amorphous", "Polycrystalline"])
+        }
+    )
+
+    predictor = SimpleMLPredictor(
+        name = "Band gap predictor",
+        description = "Predict the band gap from the chemical formula and crystallinity",
+        inputs = [
+            data_source.column_definitions["Chemical Formula"],
```

Comment:
(minor request) would be useful to explicitly state that the column_definitions are identified by the column headers, and not by the descriptor keys

Reply:
updated

Reply:
(question) is it not required to use `data_source.column_definitions` here? can you reference instead a descriptor object like `ChemicalFormulaDescriptor("formula")` ? 

also, what I meant is that it would be nice to state that the key in the `column_definitions` should refer to the csv column header and not the descriptor key (i.e. "formula")

Reply:
>  is it not required to use data_source.column_definitions here? can you reference instead a descriptor object like ChemicalFormulaDescriptor("formula") ?

It is not. You can create the descriptor directly but it needs to match the descriptor in the CSV data source, hence the comment on lines 54-55.

> what I meant is that it would be nice to state that the key in the column_definitions should refer to the csv column header and not the descriptor key (i.e. "formula")

I think that's covered in lines 40-41. "`column_definitions` maps a column header to a descriptor" states that the key in the dictionary refers to a column header. And "the column header and the descriptor key do not need to be identical" states that it doesn't refer to the descriptor key.

If you have different wording in mind, feel free to make suggested changes.

Reply:
OK, that makes sense to me. I'm good with this! Thanks